addons/osio-addon/src/main/java/io/fabric8/launcher/osio/projectiles/OsioProjectileContext.java

```diff
@@ -35,6 +35,10 @@
     @DefaultValue("1.0.0")
     private String projectVersion;
 
+    @FormParam("dependencySnapshot")
+    @DefaultValue("{}")
+    private String dependencySnapshot;
```

Comment:
Using String to represent this field may be harder to validate and parse the content. Ideally it would be nice to create Immutable VOs(using @Immutable) to define what's expected in this attribute

Reply:
If you pass only the dependencies, this can be a `String[]` containing only the dependencies in the `groupId:artifactId:version` format